### PR TITLE
OHRM5X-553: Project Related bug fixes

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains bug fixes:-
OHRM5X-1025
OHRM5X-1012
OHRM5X-1023
OHRM5X-1030
OHRM5X-1031
OHRM5X-1038

### CODE REVIEW

File: symfony/client/src/orangehrmTimePlugin/pages/project/ProjectSave.vue [hunk 1]
```diff
@@ -115,7 +116,7 @@ import ProjectAdminAutocomplete from '@/orangehrmTimePlugin/components/ProjectAd
 
 const defaultProjectModel = {
   name: null,
-  customer: null,
+  customer: {id: null, label: null},
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmTimePlugin/pages/project/ProjectSave.vue:119]: don't we need to handle this in `ProjectEdit.vue`?

File: symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php [hunk 2]
```diff
@@ -310,10 +321,17 @@ protected function getCommonBodyValidationRules(): array
 
     /**
      * @inheritDoc
+     * @throws BadRequestException
      */
     public function delete(): EndpointResult
     {
         $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_IDS);
+        foreach ($ids as $id) {
+            $hasTimesheetItems = $this->getProjectService()->getProjectDao()->hasTimesheetItems($id);
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php:330]: This need to handle in FE side as well like we handled in Reporting Method

File: symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php [hunk 3]
```diff
@@ -197,4 +200,13 @@ public function getProjectIdListForProjectAdmin(int $projectAdminEmpNumber, bool
         $result = $q->getQuery()->getArrayResult();
         return array_column($result, 'id');
     }
+
+
+    public function hasTimesheetItems(int $id): bool
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php:205]: doc comment
```suggestion
    public function hasTimesheetItemsForProject(int $projectId): bool
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/client/src/orangehrmTimePlugin/pages/project/ProjectSave.vue (hunk 1 above)
Comment (SandeepaOHRM) [reply symfony/client/src/orangehrmTimePlugin/pages/project/ProjectSave.vue:119]: yeah.

File: symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php (hunk 3 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php:205]: done

File: symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php (hunk 2 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php:330]: Noted

File: symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php [hunk 4]
```diff
@@ -38,12 +39,16 @@
...
+use OrangeHRM\Time\Traits\Service\CustomerServiceTrait;
 use OrangeHRM\Time\Traits\Service\ProjectServiceTrait;
+use OrangeHRM\Time\Traits\Service\TimesheetServiceTrait;
 
 class ProjectAPI extends Endpoint implements CrudEndpoint
 {
     use ProjectServiceTrait;
     use UserRoleManagerTrait;
+    use CustomerServiceTrait;
+    use TimesheetServiceTrait;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/ProjectAPI.php:51]: don't see usage of `TimesheetServiceTrait`

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**